rxkeystore/src/androidTest/java/com/nexenio/rxkeystore/provider/BaseCryptoProviderTest.java

```diff
@@ -19,28 +19,44 @@
 
+    protected static Provider originalProvider;
+    protected static int originalPosition;
+
     protected Context context;
     protected RxKeyStore keyStore;
     protected RxCryptoProvider cryptoProvider;
 
     @CallSuper
     protected void setUpBeforeEachTest() {
-        setupSecurityProviders();
-
         context = InstrumentationRegistry.getInstrumentation().getTargetContext();
         keyStore = createKeyStore();
         cryptoProvider = createCryptoProvider(keyStore);
     }
 
-    protected void setupSecurityProviders() {
-        final Provider provider = Security.getProvider(BouncyCastleProvider.PROVIDER_NAME);
-        if (!(provider instanceof BouncyCastleProvider)) {
+    public static void setupSecurityProviders() {
+        Provider[] providers = Security.getProviders();
+        for (int i = 0; i < providers.length; i++) {
+            Provider provider = providers[i];
+            if (BouncyCastleProvider.PROVIDER_NAME.equals(provider.getName())) {
+                originalProvider = provider;
+                originalPosition = i;
+            }
+        }
+        if (!(originalProvider instanceof BouncyCastleProvider)) {
             // Android registers its own BC provider. As it might be outdated and might not include
             // all needed ciphers, we substitute it with a known BC bundled in the app.
             // Android's BC has its package rewritten to "com.android.org.bouncycastle" and because
             // of that it's possible to have another BC implementation loaded in VM.
             Security.removeProvider(BouncyCastleProvider.PROVIDER_NAME);
-            Security.insertProviderAt(new BouncyCastleProvider(), 1);
+            // TODO: 03.06.21 was it intended to have it move to position 1 (so first item in the list) or was that by chance?
```

Comment:
The `BouncyCastleProvider` was at position 1 when I investigated but adding it with `Security.insertProviderAt(provider, 1)` will put it at position 0. 
Is it intended to have it further up or is that done by chance?

Reply:
The intention was to not change the default provider (which is at index 0). We had this at 0 before, but it caused some issues.

Reply:
Like I said, the defualt provider is at position 1 and not 0. Like stated in this [documentation](https://docs.microsoft.com/en-us/dotnet/api/java.security.security.insertproviderat?view=xamarin-android-sdk-9).

Reply:
Oh, I got that wrong. Maybe this was the reason for the issues. Actual documentation is [here](https://docs.oracle.com/javase/7/docs/api/java/security/Security.html#insertProviderAt(java.security.Provider,%20int)).